tests/aws/services/sqs/test_sqs_backdoor.py

```diff
@@ -154,6 +157,78 @@ def test_list_messages_as_json(self, sqs_create_queue, aws_client):
         assert "ApproximateFirstReceiveTimestamp" in attributes
         assert "SentTimestamp" in attributes
 
+    @markers.aws.only_localstack
+    def test_list_messages_with_invisible_messages(self, sqs_create_queue, aws_client):
+        queue_url = sqs_create_queue()
+
+        aws_client.sqs.send_message(QueueUrl=queue_url, MessageBody="message-1")
+        aws_client.sqs.send_message(QueueUrl=queue_url, MessageBody="message-2")
+        aws_client.sqs.send_message(QueueUrl=queue_url, MessageBody="message-3")
+
+        # check out a messages
+        aws_client.sqs.receive_message(QueueUrl=queue_url, MaxNumberOfMessages=1)
+
+        response = requests.get(
+            "http://localhost:4566/_aws/sqs/messages",
+            params={"QueueUrl": queue_url, "ShowInvisible": False},
+            headers={"Accept": "application/json"},
+        )
+        doc = response.json()
+        messages = doc["ReceiveMessageResponse"]["ReceiveMessageResult"]["Message"]
+        assert len(messages) == 2
+        assert messages[0]["Body"] == "message-2"
+        assert messages[1]["Body"] == "message-3"
```

Comment:
nit: Since we are not using a fifo queue, relying on the order of the messages could be dangerous. However, since this a synchronous @only_localstack test, we shouldn't really have any problems.
This holds for all code in the test that uses indices when accessing messages, but as I said, probably not really worth fretting over it.

Reply:
great catch! as you said i think it's fine now, probably not worth running the suite again :-)